File: 2025-05_Spring-101-Java-Maven/src/main/java/com/happn/worshop/infrastructure/UserEntity.java
package com.happn.worshop.infrastructure;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.*;

import java.util.UUID;

@Entity(name = "users")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Builder
public class UserEntity {

    @Id
    private UUID id;
    private String name;
    private String email;

}

File: 2025-05_Spring-101-Kotlin-Gradle/src/main/kotlin/com/happn/workshop/infra/UserEntity.kt
package com.happn.workshop.infra

import com.happn.workshop.domain.User
import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.util.UUID

@Entity(name = "users")
data class UserEntity(
    @Id
    val id: UUID,
    val name: String,
    val email: String
)


fun UserEntity.toDomain(): User = User(
    id = this.id,
    name = this.name,
    email = this.email
)

fun User.toEntity(): UserEntity = UserEntity(
    id = this.id,
    name = this.name,
    email = this.email
)